### PR TITLE
 amp-twitter: tighten up fallback logic so it does not pick up random numbers.

### PR DESCRIPTION
**Pull request by aghassemi**

### CODE REVIEW

File: build-system/global-configs/prod-config.json [hunk 1]
```diff
@@ -1,34 +1,23 @@
 {
-  "allow-doc-opt-in": ["visibility-v3", "amp-animation", "ampdoc-shell", "amp-ima-video"],
-  "allow-url-opt-in": ["pump-early-frame"],
+  "allow-doc-opt-in": ["visibility-v2"],
```
Comment (aghassemi) [inline build-system/global-configs/prod-config.json]: No idea why this is showing up. rebasing

File: 3p/twitter.js [hunk 2]
```diff
@@ -114,7 +114,12 @@ export function cleanupTweetId_(tweetid) {
     }
   }
   // 2)
-  // Otherwise just return the first number we find.
-  // Tweet Ids are always a Int64 number.
-  return tweetid.match(/\d+/)[0];
+  // Handle malformed ids such as
+  // 585110598171631616?ref_src
+  const match = tweetid.match(/^(\d+)\?ref.*/);
```
Comment (honeybadgerdontcare) [inline 3p/twitter.js]: Any reason it's `?ref` and not `?ref_src=twsrc`

Comment (aghassemi) [reply 3p/twitter.js]: just trying to cover more cases. don't think make a different at the end.

**Pull request merged**